skrub/_joiner.py

```diff
@@ -318,52 +316,47 @@ def transform(self, X, y=None):
 
         Parameters
         ----------
-        X : :obj:`~pandas.DataFrame`, shape [n_samples, n_features]
+        X : dataframe
             The main table, to be joined to the auxiliary ones.
         y : None
             Unused, only here for compatibility.
 
         Returns
         -------
-        :obj:`~pandas.DataFrame`
+        dataframe
             The final joined table.
         """
         del y
         check_is_fitted(self, "vectorizer_")
-        input_is_polars = is_polars(X)
-        X = self._check_dataframe(X)
+        X = self._main_check_input.transform(X)
         _join_utils.check_missing_columns(X, self._main_key, "'X' (the main table)")
         _join_utils.check_column_name_duplicates(
             X, self._aux_table, self.suffix, main_table_name="X"
         )
         main = self.vectorizer_.transform(
-            X[self._main_key].set_axis(self._aux_key, axis="columns")
+            ns.set_column_names(s.select(X, s.cols(*self._main_key)), self._aux_key)
+        )
+        aux_table = ns.reset_index(
+            _join_utils.add_column_name_suffix(self._aux_table, self.suffix)
         )
-        match_result = self._matching.match(main, self.max_dist_)
-        aux_table = _join_utils.add_column_name_suffix(
-            self._aux_table, self.suffix
-        ).reset_index(drop=True)
-        matching_col = match_result["index"].copy()
-        matching_col[~match_result["match_accepted"]] = -1
+        _match_result = self._matching.match(main, self.max_dist_)
```

Comment:
why do you need to rename `_match_result`?

Reply:
In a previous commit I was converting it to a dataframe, I don't need to rename :)